### PR TITLE
Do not crash on long linker commandline argument.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/1717

### CODE REVIEW

**Comment by pirapira** [comment]
Please rebase.

**Comment by chriseth** [comment]
Rebased.

**Comment by pirapira** [comment]
Why not disallow long names, and keep to well-defined cases on all major file systems?

**Comment by chriseth** [comment]
@pirapira Different systems disagree on what "long" means. Also, this bug occurs where the compiler checks if a string that might consist of multiple paths is actually an existing file.

**Comment by chriseth** [comment]
To clarify the previous (deleted) comment: You can either give these options directly on the command line or you can specify a file that contains these options. The compiler first checks whether the file exists and if it does not exist, it takes the string literally. So this is more or less unrelated to the maximum permitted length of a library - the name of the pull request could be improved :-)

File: solc/CommandLineInterface.cpp [hunk 1]
```diff
@@ -420,7 +420,16 @@ void CommandLineInterface::readInputFilesAndConfigureRemappings()
 bool CommandLineInterface::parseLibraryOption(string const& _input)
 {
 	namespace fs = boost::filesystem;
-	string data = fs::is_regular_file(_input) ? contentsString(_input) : _input;
+	string data = _input;
+	try
+	{
+		if (fs::is_regular_file(_input))
```
Comment (pirapira) [inline solc/CommandLineInterface.cpp:426]: Does this mean, by `touch`ing the file named `_input`, you can change the behavior of the compiler?  Is one able to invoke the Solidity compiler with the same arguments repeatedly, to perform stateful computation?

Comment (chriseth) [reply solc/CommandLineInterface.cpp:426]: Yes, by creating such a file, you change the behaviour of the compiler. In general, creating or modifying files whose names are mentioned on the commandline changes the behaviour of the compiler.

Comment (chriseth) [reply solc/CommandLineInterface.cpp:426]: In the case of library addresses / linker information, the name of the file has to include a `=`, though.

Comment (chriseth) [reply solc/CommandLineInterface.cpp:426]: Oh and please not that all of this is deprecated by the json-io.

Comment (pirapira) [reply solc/CommandLineInterface.cpp:426]: Ah, OK.  If it's not long-standing, no point of arguing much.

**Review by pirapira — APPROVED** [review]
Looks OK.

**Pull request merged**